Swabbr.Core/Services/UserSelectionService.cs

```diff
@@ -8,41 +8,44 @@
 using System.Collections.Generic;
 using System.Data.HashFunction.MurmurHash;
 using System.Text;
+using System.Threading.Tasks;
 
 namespace Swabbr.Core.Services
 {
     /// <summary>
     ///     Handles our hash distribution for user selection.
     /// </summary>
-    public sealed class HashDistributionService : IHashDistributionService
+    public sealed class UserSelectionService : IUserSelectionService
     {
-        private readonly SwabbrConfiguration config;
+        private readonly IUserService _userService;
+        private readonly SwabbrConfiguration _options;
         private static readonly IMurmurHash3 hasher = MurmurHash3Factory.Instance.Create();
         private static readonly UTF8Encoding encoder = new UTF8Encoding();
 
-        public uint ValidMinutes => config.VlogRequestEndTimeMinutes - config.VlogRequestStartTimeMinutes;
+        public uint ValidMinutes => _options.VlogRequestEndTimeMinutes - _options.VlogRequestStartTimeMinutes;
 
         /// <summary>
         /// Constructor for dependency injection.
         /// </summary>
-        public HashDistributionService(IOptions<SwabbrConfiguration> options)
+        public UserSelectionService(IUserService userService,
+            IOptions<SwabbrConfiguration> options)
         {
+            _userService = userService ?? throw new ArgumentNullException(nameof(userService));
             if (options == null || options.Value == null) { throw new ArgumentNullException(nameof(options)); }
             options.Value.ThrowIfInvalid();
-            config = options.Value;
+            _options = options.Value;
         }
 
         /// <summary>
-        /// Uses hash distribution to get all users which trigger in a given moment.
+        ///     Uses hash distribution to get all users 
+        ///     which trigger in a given moment.
         /// </summary>
-        /// <param name="users"><see cref="SwabbrUser"/></param>
         /// <param name="time"><see cref="DateTimeOffset"/></param>
         /// <param name="offset"><see cref="TimeSpan"/></param>
         /// <returns><see cref="SwabbrUser"/> collection</returns>
-        public IEnumerable<SwabbrUser> GetForMinute(IEnumerable<SwabbrUser> users, DateTimeOffset time, TimeSpan? offset = null)
+        public async Task<IEnumerable<SwabbrUser>> GetForMinuteAsync(DateTimeOffset time, TimeSpan? offset = null)
         {
-            if (users == null) { throw new ArgumentNullException(nameof(users)); }
-            if (time == null) { throw new ArgumentNullException(nameof(time)); }
+            var users = await _userService.GetAllVloggableUsersAsync().ConfigureAwait(false);
 
             var day = new DateTime(time.Year, time.Month, time.Day);
```

Comment:
This is the point where you add a few lines of comments. What is happening here, how does the algo work? Maybe move the comment from below to here?

Reply:
Agreed